deliveries/typechain-polkadot-milestone-1.md

```diff
@@ -0,0 +1,24 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).**
+
+* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/typechain-polkadot.md
+* **Milestone Number:** 1
+
+**Context** (optional)
+First Milestone for Typechain-Polkadot includes parsing types, generating output, examples, and testing.
+To parse types, we have investigated how the Polkadot-js library uses types. So we have come to make a recursive algorithm that generates all types before using them. When the parser gets a request to generate a type, it first checks the type' type and then decides what to do next. Ink! Smart contracts can produce Composite, Variant, Sequence, Array, Tuple, Primitive, and we have a different approach for every type.
+
+**Deliverables**
+
+| Number | Deliverable | Link | Notes |
+| ------------- | ------------- | ------------- |------------- |
+| 0a | License | https://github.com/Supercolony-net/typechain-polkadot/blob/master/LICENSE | MIT License |
+| 0b | Documentation | https://github.com/Supercolony-net/typechain-polkadot/tree/master/docs | We will add more documentation and examples in the second milestone with covering upcoming features and explaination of typechain logic. |
+| 1 | TS types | https://github.com/Supercolony-net/typechain-polkadot/blob/master/src/index.ts | While testing everything, we figured out that types for arguments and types for return values are not the same; for example, if we have u128 as an argument, we can pass number, string, BN, bigint, but polkadot.js if it's return type can return only number or string, so we are processing types depending on its usage. |
+| 2 | Runtime code | https://github.com/Supercolony-net/typechain-polkadot/tree/master/src/templates | We use the Handlebars library to quickly generate output files with template format (src/templates). |
+| 3 | Testing | https://github.com/Supercolony-net/typechain-polkadot/tree/master/tests | Covered psp22, psp34 and enums usage.
+| 4 | NPM Packaging | https://github.com/Supercolony-net/typechain-polkadot/packages/1500761 | In the 2nd Milestone we will have two packages for typechain: typechain and typchain-parser. |
+
+
+**Additional Information**
```

Comment:
The reviewer of the report should be clear on how to test typechain. Maybe it will be good to highlight here some examples and put the link to setup instructions. You can try to give that instruction to someone in our team first to be sure that everything is clear=)

Reply:
Ok, thanks